galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/CPSRoute.java

```diff
@@ -121,26 +84,28 @@ protected boolean checkRequestHasContent(HttpServletRequest request) throws Inte
      * Returns a single property from a given namespace.
      * If the namespace provided is hidden, does not exist or has no matching property, it returns null
      * If the namespace provided does not match any existing namepsaces an exception will be thrown
-     * @param namespace
+     * @param namespaceName
      * @param propertyName
-     * @return Map.Entry<String, String> 
+     * @return Map.Entry of String, String 
+     * @throws ConfigurationPropertyStoreException
      * @throws FrameworkException
      */
-    protected Map.Entry<String, String> retrieveSingleProperty(String namespace, String propertyName) throws  InternalServletException {
-        if (isHiddenNamespace(namespace)){
-            ServletError error = new ServletError(GAL5016_INVALID_NAMESPACE_ERROR,namespace);  
+    protected Map.Entry<String, String> retrieveSingleProperty(String namespaceName, String propertyName) throws  InternalServletException, ConfigurationPropertyStoreException {
+        Namespace namespace = new Namespace(namespaceName);
+        if (namespace.isHiddenNamespace()){
+            ServletError error = new ServletError(GAL5016_INVALID_NAMESPACE_ERROR,namespace.getName());  
             throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
         }
         try{
-            Map<String, String> properties = getAllProperties(namespace);
+            Map<String, String> properties = getAllProperties(namespace.getName());
```

Comment:
We already have namespaceName, so we could reuse that instead of repeatedly getting the namespace's name - not sure if we need to create a new Namespace object when getting a property.